src/cmark.h

```diff
@@ -558,6 +558,9 @@ void cmark_parser_free(cmark_parser *parser);
 CMARK_EXPORT
 void cmark_parser_feed(cmark_parser *parser, const char *buffer, size_t len);
 
+CMARK_EXPORT
+void cmark_parser_feed_reentrant(cmark_parser *parser, const char *buffer, size_t len);
+
 /** Finish parsing and return a pointer to a tree of nodes.
  */
 CMARK_EXPORT
```

Comment:
It isn't directly, though I can't imagine another use case for it. Here's why I need it for reference:

https://github.com/hotdoc/hotdoc/blob/master/hotdoc/parsers/cmark_include_extension.c#L80

This allows me to offer a context-aware / recursive inclusion syntax, that a preprocessor could not. It was how I implemented it a first but things like:

```
\```
{{ do_not_try_to_include_this_really_we_are_in_a_code_block_stop_embarassing_yourself }}
\```
```

could potentially break. By the way formatting this example makes me wonder how we're supposed to escape the code block open / code block close patterns, but that's a completely different issue :)